src/modules/services/submodule/settings/components/changeAccountPassword.vue

```diff
@@ -0,0 +1,132 @@
+  import PasswordInput from '@/components/PasswordInput.vue';
+  import { computed, ref } from 'vue';
+  import { networkState } from "@/state/networkState";
+  import { walletState } from '@/state/walletState';
+  import { WalletUtils } from '@/util/walletUtils';
+  import { useI18n } from 'vue-i18n';
+  import { AppState } from '@/state/appState';
+  import { WalletAccount } from "@/models/walletAccount"
+  import { useRouter } from "vue-router";
+  export default{
+    name: 'changeAccountPassword',
+    components:{
+      PasswordInput
+    },
+    setup(){
+      const {t} = useI18n();
+      const toggleModal = ref(false);
+      const toggleChangePasswrd= ref(false)
+      const showPasswdError = ref(false);
+      const passwdPattern = "^[^ ]{8,}$";
+      const router = useRouter();
+      let err = ref('');
+      let walletPasswd = ref('');
+      let passwd = ref('');
+      const confirmPasswd = ref("");
+
+      const disableShow = computed(() => !(walletPasswd.value.match(passwdPattern)));
+
+      const showConfirmPasswdError = computed(
+        () => (confirmPasswd.value != walletPasswd.value && confirmPasswd.value != '')
+      );
+
+    const clearInput = () => {
+      walletPasswd.value = "";
+      confirmPasswd.value = "";
+    };
+
+    const logout = () => {
+      walletState.isLogin = false;
+      router.push({ name: "Home"});
+    };
+
+      const verifyWalletPwPk =()=> {
+        if (WalletUtils.verifyWalletPassword(walletState.currentLoggedInWallet.name,networkState.chainNetworkName,walletPasswd.value)) {
+          // pw is correct
+          toggleChangePasswrd.value = true
+          passwd.value=walletPasswd.value
+          clearInput()
+          err.value=''
+        } else {
+          let walletName = walletState.currentLoggedInWallet.name
+          err.value = t('general.walletPasswordInvalid',{name: walletName});
+        } 
+     }
+
+     const changeWalletPasswd = ()=> {
+    for(let i = 0; i<walletState.currentLoggedInWallet.accounts.length;i++){
+      let currentAccount=walletState.currentLoggedInWallet.accounts[i]
+      const passwordInstance = WalletUtils.createPassword(passwd.value);
+      const walletPrivateKey = WalletUtils.decryptPrivateKey(passwordInstance,currentAccount.encrypted,currentAccount.iv);
+      let password = WalletUtils.createPassword(walletPasswd.value);
+      const wallet = WalletUtils.createAccountSimpleFromPrivateKey(currentAccount.name, password, walletPrivateKey, AppState.networkType);
+      let walletAccount = new WalletAccount(currentAccount.name, currentAccount.publicKey, wallet.publicAccount.address.plain() , "pass:bip32",  wallet.encryptedPrivateKey.encryptedKey, wallet.encryptedPrivateKey.iv);
+      walletState.currentLoggedInWallet.accounts[i].encrypted= walletAccount.encrypted
+      walletState.currentLoggedInWallet.accounts[i].iv= walletAccount.iv
+    }
+    walletState.wallets.saveMyWalletOnlytoLocalStorage(walletState.currentLoggedInWallet)
```

Comment:
maybe show some message to user that password has changed, or user might not know what happened.

Reply:
Noted